### PR TITLE
Use 3 argument Array.Copy, as it is more optimized in .NET5

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

**Comment by jkotas** [comment]
ImmutableArray tests are failing:
```
    System.Collections.Immutable.Tests.ImmutableArrayTest.CopyToInvalid(source: []) [FAIL]
      Assert.Equal() Failure
                     (pos 4)
      Expected: dest
      Actual:   destinationArray
                     (pos 4)
      Stack Trace:
        /_/src/libraries/Common/tests/CoreFx.Private.TestUtilities/System/AssertExtensions.cs(46,0): at System.AssertExtensions.Throws[T](String netCoreParamName, String netFxParamName, Action action)
        /_/src/libraries/System.Collections.Immutable/tests/ImmutableArrayTest.cs(1635,0): at System.Collections.Immutable.Tests.ImmutableArrayTest.CopyToInvalid(IEnumerable`1 source)
    System.Collections.Immutable.Tests.ImmutableArrayTest.CopyToInvalid(source: [1]) [FAIL]
```

**Comment by jkotas** [comment]
It should be fine to fix the test failures by relaxing the argument name validation in this test.

**Comment by Therzok** [comment]
@jkotas Do you mean something like:

```csharp
// Skip checking .NETFramework parameter name is either 'destinationArray' or 'dest'
AssertExtensions.Throws<ArgumentNullException>("destinationArray", netFxParamName: null, () => array.CopyTo(null));
AssertExtensions.Throws<ArgumentNullException>("destinationArray", netFxParamName: null, () => array.CopyTo(null, 0));
AssertExtensions.Throws<ArgumentNullException>("destinationArray", netFxParamName: null, () => array.CopyTo(0, null, 0, 0));
AssertExtensions.Throws<ArgumentNullException>("destinationArray", netFxParamName: null, () => array.CopyTo(-1, null, -1, -1)); // The destination should be validated first.
```

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Collections.Immutable/tests/ImmutableArrayTest.cs [hunk 1]
```diff
@@ -1631,10 +1631,11 @@ public void CopyToInvalid(IEnumerable<int> source)
             // ImmutableArray<T>.CopyTo defers to Array.Copy for argument validation, so
             // the parameter names here come from Array.Copy.
 
-            AssertExtensions.Throws<ArgumentNullException>("destinationArray", "dest", () => array.CopyTo(null));
-            AssertExtensions.Throws<ArgumentNullException>("destinationArray", "dest", () => array.CopyTo(null, 0));
-            AssertExtensions.Throws<ArgumentNullException>("destinationArray", "dest", () => array.CopyTo(0, null, 0, 0));
-            AssertExtensions.Throws<ArgumentNullException>("destinationArray", "dest", () => array.CopyTo(-1, null, -1, -1)); // The destination should be validated first.
+            // Skip checking .NETFramework parameter name is either 'destinationArray' or 'dest'
+            AssertExtensions.Throws<ArgumentNullException>("destinationArray", netFxParamName: null, () => array.CopyTo(null));
```
Comment (jkotas) [inline src/libraries/System.Collections.Immutable/tests/ImmutableArrayTest.cs]:
```suggestion
            AssertExtensions.Throws<ArgumentNullException>("destinationArray", "dest", () => array.CopyTo(null));
```
This is the style used in all other places that compensate for this problem. Also, the comment is unnecessary - none of the other places have it.

**Comment by stephentoub** [comment]
@Therzok, still working on this?

**Comment by Therzok** [comment]
@stephentoub I'll setup the repo on a Windows machine to see why it's failing, the tests passed locally on OS X. Sorry for the delays.

**Comment by stephentoub** [comment]
If you need help, let me know (and if I have push permissions to your branch, I can just fix it for you, too).

**Comment by Therzok** [comment]
I've had a few hoops and hurdles. If you have MSBuild Build Tools installed and they don';t have the required pre-reqs, you could end up opening the wrong developer command prompt to build. I'm trying to get the in-VS tests to run right now, looking at the test explorer, it doesn't seem to run them. Although it seems the top menu run seems to work.

**Comment by Therzok** [comment]
Okay, so, I've managed to get it setup and run it. Whenever I try to run the .NET Framework config, I'm getting `Illegal characters in path.`, with no extra explanation.

**Comment by stephentoub** [comment]
I fixed up the commit.  Thanks for the contribution.

**Comment by Therzok** [comment]
Thanks for that!

**Pull request merged**